### PR TITLE
Mining Overhaul 2.0, Chapter 1: Abyss

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Well, I decided to atomise my old mining overhaul while fixing some stuff and adding new features. This is the first part(there will be around 3) which features new Z level - Abyss. You can get to it by either falling into the chasm or killing Legion. Exiting the Abyss requires Legion to be killed because the portal that leads you out is located right behind him. There are 3 new mobs featured: Abyss Demon, which is basically a basilisk and ice demon mix, fire wisp which is a suicidal enemy that will rush you, and Ash Whelps, small drakes that can breath fire and swoop up just like their parents, but on shorter distances. Oh yeah, and goliaths located in the Abyss will be able to shoot their tendrils like projectiles, getting their targets closed and allowing them to land a hit.

## Why It's Good For The Game

Getting instakilled by chasms was really lame - now they just send you to dangerous as fuck Z level.

## Changelog
:cl:
add: Added a new mining Z level - Abyss
add: Added 3 new hostile mobs: Abyss demon, Fire wisp and Ash whelp
tweak: Chasms now send you to Abyss instead of instakilling
tweak: Goliaths now can shoot their tentacles if they're located in the Abyss
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/ash_whelp.dm [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	deathmessage = "collapses on it's side."
+
+	var/swooping = FALSE
+
+/mob/living/simple_animal/hostile/asteroid/ice_whelp/lavaland/OpenFire()
+	if(swooping)
+		return
+	var/swoop_chance = clamp(((maxHealth - health)/25), 0, 20)
+
+	if(prob(swoop_chance) && target in range(4, src))
```
Comment (SandPoot) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/ash_whelp.dm]:
```suggestion
	if(prob(swoop_chance) && (target in range(4, src)))
```

**Comment by YakumoChen** [comment]
Ehhhh.... seems like a lot of work for a whole extra Z level that amounts to either Permadeath with extra steps (killing Legion is basically end game for mining) or obstanately just adding another literal layer of powergame for miners intentionally jumping in chasms and getting out with one of the several items that teleport you to safety. One of which miners can get at round start.

**Comment by ghost** [comment]
> Ehhhh.... seems like a lot of work for a whole extra Z level that amounts to either Permadeath with extra steps (killing Legion is basically end game for mining) or obstanately just adding another literal layer of powergame for miners intentionally jumping in chasms and getting out with one of the several items that teleport you to safety. One of which miners can get at round start.

yeah not to mention they're trying to do this retarded shit again where they steal someone else's code and say its their's
some people live in fucking stasis mentally i think

**Comment by SmArtKar** [comment]
> > Ehhhh.... seems like a lot of work for a whole extra Z level that amounts to either Permadeath with extra steps (killing Legion is basically end game for mining) or obstanately just adding another literal layer of powergame for miners intentionally jumping in chasms and getting out with one of the several items that teleport you to safety. One of which miners can get at round start.
> 
> yeah not to mention they're trying to do this retarded shit again where they steal someone else's code and say its there's
> some people live in fucking stasis mentally i think

please stop being toxic as fuck

**Comment by DeltaFire15** [comment]
Merge conflicting.

**Comment by DeltaFire15** [comment]
This has been in merge conflict purgatory since April. Closing this PR.

**Pull request closed without merging**